src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/ApartmentValueFunction.java

```diff
@@ -685,91 +601,115 @@ public static ApartmentValueFunction getRandomApartmentValueFunction() {
 	 * would be a discrete criteria (e.g. the number of bedrooms)
 	 * 
 	 * The goal is to replace a LinearValueFunction's bound by a new bound
+	 * Warning : The values of the object should be instanciate before using this function 
+	 * or an error will appear
 	 * 
-	 * @param criteria
-	 * @param newBound
-	 * @param lower    is true when we want to adapt the lower bound, false if we
-	 *                 want to adapt the upper bound
+	 * @param criterion the criterion to adapt. This criterion should not be a
+	 *                  boolean as TV for example.
+	 * @param newBound  the new bound to define
+	 * @param lower     is true when we want to adapt the lower bound, false if we
+	 *                  want to adapt the upper bound
+	 * @return an object ApartmentValueFunction
 	 */
 	public ApartmentValueFunction adaptBounds(Criterion criterion, double newBound, boolean lower) {
 
-		ApartmentValueFunction avf = cloneAVF();
```

Comment:
Je préférais l’architecture précédente, qui s’approchait d’une AVF immuable. Une AVF immuable permet de mieux contrôler les changements. (Voir aussi liens concernant l’immuabilité dans BP Local design.) On va dans le mauvais sens en modifiant les bornes sur place. (FUTURE)

Reply:
Opening an issue concerning this comment [here](https://github.com/av1m/Apartments/issues/22)
This can be fix on iteration 2 @GabG02

Reply:
Fixed in commit 177789f5dd6d5809d8dd9906edf5e3b460a624c1